src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+
+    # Setting the log output as Empty
+    diagnoser_container_log = ""
+    yaml_file_path = os.path.join(absolute_path,"troubleshoot_diagnoser_job.yaml")
+    command = ["kubectl", "delete", "-f", ""]
+    command[3]=str(yaml_file_path)
+
+    # To handle the user keyboard Interrupt
+    try:
+
+        # Executing the diagnoser_job.yaml
+        config.load_kube_config()
+        k8s_client = client.ApiClient()
+
+        try:
+            utils.create_from_yaml(k8s_client, yaml_file_path)
+        except Exception as e:
+            pass
+
+        # Watching for diagnosercontianer to reach in completed stage
+        w = watch.Watch()
+        counter = 0
+        for event in w.stream(batchv1_api_instance.list_namespaced_job, namespace=namespace, label_selector="", timeout_seconds=90):
+            try:
+                if event["object"].metadata.name == "diagnosercontainer" and event["object"].status.conditions[0].type == "Complete":
+                    counter = 2
+                    w.stop()
+
+                elif event["object"].metadata.name == "diagnosercontainer" and event["object"].status.conditions[0].type == "Pending":
+                    counter = 1
+
+            except Exception as e:
+                continue
+            else:
+                continue
+
+        # If container not created then clearing all the reosurce with proper error message
+        if (counter == 0):
+            logger.warning("Unable to create the diagnoser job in the cluster. It may be caused due to insufficient resource availability on the cluster. ")
+            subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        
+        # If container stuck in pending state then clearing all the resources with proper error message
+        elif (counter == 1):
+            logger.warning("The diagnoser job is stuck in the 'Pending' state in the cluster. It may be caused due to insufficient resource availability on the cluster. ")
+            subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+        else:
+
+            # Fetching the Diagnoser Container logs
+            try:
+
+                job_name = "diagnosercontainer"
+
+                all_pods = corev1_api_instance.list_namespaced_pod(namespace)
+                # Traversing thorugh all agents
+                for each_pod in all_pods.items:
+
+                    # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                    pod_name = each_pod.metadata.name
+
+                    if(pod_name.startswith(job_name)):
+
+                        # Creating a text file with the name of the container and adding that containers logs in it
+                        diagnoser_container_log = corev1_api_instance.read_namespaced_pod_log(name=pod_name, container="networktest1", namespace=namespace)
```

Comment:
rename the container to something more descriptive. May be "outboundconnectivitytest"

Reply:
Renamed to azure-arc-diagnoser-container